### ISSUE TITLE
H110i Fan 0 data also shown as Pump Data

### ISSUE THREAD

**Issue by rigred** [audiohacked/OpenCorsairLink#137]
**Describe the bug**
On the H110i I have the Pump data is not displayed, instead the First Fan data is repeated as the Pump data. 
No pump state changes can be seen. But it seems that:
* Pump speeds can be set. 
* Led colors can be changed. 
* Fan speeds can be changed.

**To Reproduce**
1. Run `OpenCorsairLink --device 0`
2. View the Output
3. Confirm it is true by disconnecting Fan 0
4. Notice that the Pump speed information also falls to 0

**Expected behavior**
The Pump speed data is shown and not Fan 0 data.

**Screenshots**
Here I have disconnected Fan 0 to show that indeed the Fan 0 data is being shown as Pump data.
![screenshot_20180923_095336](https://user-images.githubusercontent.com/5716332/45925544-db590f00-bf17-11e8-8155-a8ef50f5a464.jpeg)

**System:**
 - Arch Linux 
 - Kernel 4.18.9
 - Both OpenCorsairLink master & testing build
 -  [H110i Cooler](https://www.corsair.com/ww/en/Categories/Products/Liquid-Cooling/Dual-Radiator-Liquid-Coolers/Hydro-Series%E2%84%A2-H110i-280mm-Extreme-Performance-Liquid-CPU-Cooler/p/CW-9060026-WW)

**Additional context**
`--debug --dump`
```text
No (device_id 0x42)
Corsair product detected. Checking if device is H110i... 
---- Packet dump: -----------------------------
8a 07 42 00 00 00 00 00 00 00 00 00 00 00 00 00 
00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
-----------------------------------------------
Dev=0, CorsairLink Device Found: H110i!
Checking USB device 7 (1d6b:0002)...

```

USB Device dump
```
Bus 001 Device 003: ID 1b1c:0c04 Corsair Link Cooling Node
Device Descriptor:
  bLength                18
  bDescriptorType         1
  bcdUSB               2.00
  bDeviceClass            0 
  bDeviceSubClass         0 
  bDeviceProtocol         0 
  bMaxPacketSize0        64
  idVendor           0x1b1c Corsair
  idProduct          0x0c04 Link Cooling Node
  bcdDevice            2.00
  iManufacturer           1 Corsair Memory, Inc.
  iProduct                2 Integrated USB Bridge
  iSerial                 0 
  bNumConfigurations      1
  Configuration Descriptor:
    bLength                 9
    bDescriptorType         2
    wTotalLength       0x0022
    bNumInterfaces          1
    bConfigurationValue     1
    iConfiguration          0 
    bmAttributes         0x80
      (Bus Powered)
    MaxPower              100mA
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        0
      bAlternateSetting       0
      bNumEndpoints           1
      bInterfaceClass         3 Human Interface Device
      bInterfaceSubClass      0 
      bInterfaceProtocol      0 
      iInterface              0 
        HID Device Descriptor:
          bLength                 9
          bDescriptorType        33
          bcdHID               1.11
          bCountryCode            0 Not supported
          bNumDescriptors         1
          bDescriptorType        34 Report
          wDescriptorLength      37
         Report Descriptors: 
           ** UNAVAILABLE **
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x81  EP 1 IN
        bmAttributes            3
          Transfer Type            Interrupt
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0040  1x 64 bytes
        bInterval               1
can't get device qualifier: Resource temporarily unavailable
can't get debug descriptor: Resource temporarily unavailable
Device Status:     0x0080
  (Bus Powered)
```

**Comment by rigred** [comment]
If I want to correct this, code you give me pointer at where in the code to check?

**Comment by rigred** [comment]
I've seen a few other (incomplete) issues by others that confirm the issue is correct.

https://github.com/audiohacked/OpenCorsairLink/issues/102
https://github.com/audiohacked/OpenCorsairLink/issues/136

Otherwise Related:
https://github.com/audiohacked/OpenCorsairLink/issues/65

**Comment by hoanguc3m** [comment]
> If I want to correct this in the code, could you give me pointer at where to check?

My solution: I install virtual box with windows and install Corsair link. Then, use cooler as usb and change the configure.
If that could help. Bests.

**Comment by audiohacked** [comment]
@rigred Check that the pump index is correct in `device.c`
Also insure the logic for the water cooler is correct in `logic/settings/hydro_coolit.c`

**Comment by audiohacked** [comment]
@rigred I just made a minor fix in `device.c` for pump indexes for a few devices.

**Comment by rigred** [comment]
I've tested the code and it doesn't change anything functionally. 
I know you incremented the `pump_index` values for coolit devices by 1.
But everything is still reporting the same values.

Note how Pump data is showing the same values as Fan 0.
```
sudo ./OpenCorsairLink.elf --device 0
Dev=0, CorsairLink Device Found: H110i!

Vendor: Corsair
Product: H110i
Firmware: 2.0.0
Temperature 0: 29.12 C
Fan 0:  Quiet Mode (4PIN)
        Current/Max Speed 1050/2061 RPM
Fan 1:  Quiet Mode (4PIN)
        Current/Max Speed 1031/2089 RPM
Pump:   Mode 0x89
        Current/Max Speed 1050/2061 RPM
```
Fan 1 is actually reporting correct information, Fan 0 is also correct.

I haven't yet taken the time to fully immerse myself in the codebase ,but I have looked in the places you told me and there it looks ok. 
I suspect that somewhere the pump data is being overwritten with Fan data or a pointer is wrong.
I will continue to investigate.

**Comment by audiohacked** [comment]
@rigred As a note, the pump functions are in `protocol/coolit/pump.c`.

**Comment by NoMod-Programming** [comment]
I've submitted a fix for this in #144. Note that I'm not sure if the pump data is correct for other models, only that it works on the H110i. If someone using another pump could test it out, that'd be great!

**Comment by audiohacked** [comment]
Fixed!
